airflow/providers/cncf/kubernetes/decorators/kubernetes.py

```diff
@@ -57,7 +57,7 @@ def _read_file_contents(filename):
 class _KubernetesDecoratedOperator(DecoratedOperator, KubernetesPodOperator):
     custom_operator_name = "@task.kubernetes"
 
-    template_fields: Sequence[str] = ("op_args", "op_kwargs")
+    template_fields: Sequence[str] = ("op_args", "op_kwargs", "env_vars")
```

Comment:
what about 
```python
template_fields: Sequence[str] = tuple(
        {"op_args", "op_kwargs"'} | set(KubernetesPodOperator.template_fields)
    )
    
``` ?

Reply:
Unfortunately not all of those `template_fields` are applicable. Specifically `cmds` and `arguments` are used to run the kuberenetes pod using the python code written under the decorator

https://github.com/apache/airflow/blob/64174ce25ae800a38e712aa0bd62a5893ea2ff99/airflow/providers/cncf/kubernetes/decorators/kubernetes.py#L66-L74

If you pass in the `cmds` attribute, it raises an error. The list could be expanded to be: `image`, `env_vars`, `labels`, `config_file`, `pod_template_file`, and `namespace`, as described [here](https://docs.astronomer.io/learn/kubepod-operator#optional-arguments)

Reply:
```Python
template_fields: Sequence[str] = tuple(
    {"op_args", "op_kwargs"} | set(KubernetesPodOperator.template_fields) - {"cmds", "arguments"}
)
```
Would work though.

Reply:
LGTM, could use a comment though